// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";


package bankworld;

// The customer service definition.
service Customer {
  rpc createStub (StubEvent) returns (StubDone) {}
  rpc executeEvents (StubEvent) returns (StubDone) {}
}


service Branch {
  rpc createStub (StubEvent) returns (StubDone) {}
  rpc MsgDelivery (BranchRequest) returns (BranchReply) {}
  rpc Propagate_Withdraw (WithdrawRequest) returns (WithdrawReply) {}
  rpc Propagate_Deposit (DepositRequest) returns (DepositReply) {}
}

// The request message containing the user's name.
message BranchRequest {
  string msg = 1;
}

// The response message containing the greetings
message BranchReply {
  string branch_msg = 1;
}

// The request message containing the user's name.
message WithdrawRequest {
  string msg = 1;
}

// The response message containing the greetings
message WithdrawReply {
  string withdraw_msg = 1;
}

// The request message containing the user's name.
message DepositRequest {
  string msg = 1;
}

// The response message containing the greetings
message DepositReply {
  string deposit_msg = 1;
}

message StubDone {
  string stub_msg = 1;
}

message StubEvent {
  string stub_event = 1;
}

//message Deposit_amt {
//  string number = 1;
//}

//message Deposit_result {
//  string number = 1;
//}
